create_database.py

```diff
@@ -0,0 +1,107 @@
+                           nationality TEXT,
+                           birthdate DATE)
+    """)
+    # Retrieve users data.
+    data = read_file(users_filename)
+    assert data, "There is no data in file: %s" % users_filename
+    data_chunks = chunk_data(data, 3)
+    # Insert data into the users table.
+    for chunk in data_chunks:
+        username = parse(r"username:(?P<username>[\w.-]+)", 'username', chunk[0])
+        nationality = parse("nationality:(?P<nationality>[a-zA-Z]+ ?[a-zA-Z]*)",
+                            "nationality",
+                            chunk[1],
+                           )
+        birthdate = parse("birthdate:(?P<birthdate>[0-9]{1,2}/[0-9]{1,2}/[0-9]{4})",
+                          "birthdate",
+                          chunk[2],
+                         )
+        if username and nationality and birthdate:
+            assert date_validation.is_valid_str_date(birthdate), "A valid date must be provided."
+            cursor.execute("""INSERT INTO users(username, nationality, birthdate)
+                              VALUES(?,?,?)""", (username.lower(), nationality.lower(), birthdate))
+    # Save and exit the database file.
+    database.commit()
+    database.close()
+
+def create_date_formats_table(db_filename, date_formats_filename):
+    """Create a table of date formats with columns for nationality and
+    the date format and insert the data found in the given date_formats
+    file.
+    """
+    # Open and connect to the database.
+    database = sqlite3.connect(db_filename)
+    cursor = database.cursor()
+    # Create date_formats table.
+    cursor.execute("""
+        CREATE TABLE date_formats(nationality TEXT PRIMARY KEY,
+                                  date_format TEXT)
+    """)
+    # Retrieve date_formats data.
+    data = read_file(date_formats_filename)
+    assert data, "There is no data in file: %s" % date_formats_filename
+    data_chunks = chunk_data(data, 1)
+    # Insert data into the date_formats table.
+    for chunk in data_chunks:
+        if chunk:
+            nationality = parse("(?P<nationality>[a-zA-Z]+ ?[a-zA-Z]*),",
+                                'nationality',
+                                chunk[0])
+            date_format = parse("(?P<date_format>'{[a-zA-Z]+}/{[a-zA-Z]+}/{[a-zA-Z]+}')",
+                                'date_format',
+                                chunk[0])
+            if nationality and date_format:
+                cursor.execute("""INSERT INTO date_formats(nationality, date_format)
+                                  VALUES (?,?)""", (nationality.lower(), date_format))
+    # Save and exit the database file.
+    database.commit()
+    database.close()
+
+def create_database(db_filename, users_filename, date_formats_filename):
+    """Create an SQLite database file consisting of two tables, 'users'
+    and 'date_formats'.
+    """
+    create_user_table(db_filename, users_filename)
```

Comment:
This is 'logically' cohesive rather than a higher level. Do you see why? You're basically putting these two function calls together because they both create tables, not because they have any other relation. Why should, for instance, the users filename and the dates formats filename ever need to be passed into the same function? May want to just put this behavior in your main rather than creating a function for it